src/Checkpointer.jl

```diff
@@ -59,7 +64,7 @@ function restart_model_state!(sim::Interfacer.ComponentModelSimulation, t::Int;
     @info "Setting " Interfacer.name(sim) " state to checkpoint: $input_file, corresponding to day $day second $sec"
 
     # open file and read
-    hdfreader = InputOutput.HDF5Reader(input_file)
+    hdfreader = InputOutput.HDF5Reader(input_file, comms_ctx)
```

Comment:
In general do we want to refer to these as `hdf5reader` ? Here and elsewhere - users might expect general compatibility with hdf formats otherwise.

Reply:
Sure, we were following the syntax of ClimaCore and ClimaAtmos, but we can change the naming to something more legible in the local scope of this function. Would be good to have this discussion more widely at some point 😅 how about `checkpoint_writer` and `restart_reader`?